VLApi/Controllers/BooksController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using VL.Shared.Data;
+using VL.Shared.Model;
+
+namespace VLApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class BooksController : ControllerBase
+    {
+        private readonly ApplicationDbContext _context;
+
+        public BooksController(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/Books
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Book>>> GetBook()
+        {
+            return await _context.Book.ToListAsync();
+        }
+
+        // GET: api/Books/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Book>> GetBook(int id)
+        {
+            var book = await _context.Book.FindAsync(id);
+
+            if (book == null)
+            {
+                return NotFound();
+            }
+
+            return book;
+        }
+
+        // PUT: api/Books/5
+        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutBook(int id, Book book)
```

Comment:
(int id, Book book)

You're passing in Book already and Book.Id exists you do not need id as a parameter

Reply:
Isn't the id parameter provided by the URI and then used to confirm id = book.id otherwise bad request?

Reply:
This is security through obscurity at best (really it's not even anything)

For a request to the API from swagger, curl, or the web app.... that should only be lets say updating a Book with an ID of 1...

I can just as easily send...

HTTP PUT localhost:5000 api/books/1 with a body of { Id: 1, Name: 'Title', etc.) <-- Success
HTTP PUT localhost:5000 api/books/1 with a body of { Id: 2, Name: 'Title', etc.) <-- Error
HTTP PUT localhost:5000 api/books/2 with a body of { Id: 2, Name: 'Title', etc.) <-- Success

So what does a parameter of id do for you?